d2ir/d2ir.go

```diff
@@ -762,16 +759,12 @@ func (m *Map) CreateEdge(eid *EdgeID, refctx *RefContext) (*Edge, error) {
 		return nil, d2parser.Errorf(refctx.Edge, "cannot create edge inside edge")
 	}
 
-	eid, m, err := eid.resolveUnderscores(m)
+	common, eid, m, err := eid.resolveToCommon(m)
 	if err != nil {
 		return nil, d2parser.Errorf(refctx.Edge, err.Error())
 	}
-	common, eid := eid.trimCommon()
-	if len(common) > 0 {
-		tmp := *refctx.Edge.Src
```

Comment:
This was a hack to reuse `EnsureField`, common should be reverted back to `[]string`. It doesn't reflect something in the syntax. Ideally later we add a `ensureFieldIDA` that accepts `[]string` directly.